### PR TITLE
Clean up rustdoc js testers

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
I realized after the improvement made by @ollie27 on the rustdoc-js-tester that a lot of code was actually duplicated. This PR intends to remove this duplication, making it simpler to update in case of future main.js updates.

r? @ollie27 

cc @kinnison

### CODE REVIEW

File: src/tools/rustdoc-js-std/tester.js [hunk 1]
```diff
@@ -1,236 +1,40 @@
...
+            }
+            matches.push([entry, total]);
         }
     }
-    return null;
+    if (matches.length === 0) {
+        return null;
+    }
+    // We make a reverse sort to have the "highest" file. Very useful in case you didn't clean up
+    // you std doc folder...
```
Comment (ollie27) [inline src/tools/rustdoc-js-std/tester.js]: This is a lot of code for such an edge case.

Instead, how about passing the resource suffix as an argument to `tester.js` and using that to build the filenames? That way we won't need the `findFile` or `readFileMatching` functions at all and there won't be any issues with outdated files.

**Review by ollie27 — COMMENTED** [review]
Could we combine these three files into a single `tester.js` file that works for both test suites?

File: src/tools/rustdoc-js-std/tester.js (hunk 1 above)
Comment (GuillaumeGomez) [reply src/tools/rustdoc-js-std/tester.js]: That'd make things simpler indeed!

**Comment by GuillaumeGomez** [comment]
@ollie27 I'm a bit shared on this: it'd mean adding more options, which I'm not a big fan of.

**Comment by ollie27** [comment]
> it'd mean adding more options, which I'm not a big fan of.

I don't think merging the files would, but either way it can done in a follow up PR.

**Comment by GuillaumeGomez** [comment]
Ok, opening an issue to not lose track of it then. Thanks a lot @ollie27 !

**Comment by ollie27** [comment]
https://github.com/rust-lang/rust/pull/70224#discussion_r396030330 still hasn't been resolved. I guess I should have been clearer but that was the reason I hadn't r+'d this.

**Pull request merged**